autorest/adal/token.go

```diff
@@ -709,36 +713,51 @@ func retry(sender Sender, req *http.Request) (resp *http.Response, err error) {
 		http.StatusNotExtended,
 		http.StatusNetworkAuthenticationRequired)
 
+	// see https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/how-to-use-vm-token#retry-guidance
+
+	const maxAttempts int = 5
+	const deltaDelay int = 2          // seconds
```

Comment:
I'm not sure this needs to be a constant, it just seems like it's forcing us to choose an awkward name, and it's not even really in seconds. I'd just get rid of it and use `math.Pow(2.0, float64(attempt))` below.

Reply:
I would like to avoid magic numbers.  The name comes from the spec here https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/how-to-use-vm-token#retry-guidance (agree it isn't the best but I wanted to keep it close).

Reply:
Hmm thinking about this a bit more I will remove this const and add a comment where it's being used.

Reply:
Fixed.